scripts/watch_helpers/__init__.py

```diff
@@ -22,40 +23,57 @@ def iter_matching_objects(
     for page in page_iter:
         for obj in page['Contents']:
             if include_pattern.match(obj['Key']):
-                yield obj
+                yield f's3://{bucket}/{obj["Key"]}'  # TODO: modified date
 
 
-def iter_stac_items(url: str, api_key: str = None) -> Generator[dict, None, None]:
-    stack = [url]
+def get_stac_item_self_link(links):
+    for link in links:
+        if link['rel'] == 'self':
+            return link['href']
+    raise ValueError('No self link found')
+
+
+def iter_stac_item_urls(
+    url: str, collections: List[str], min_date: datetime, max_date: datetime, api_key: str = None
+) -> Generator[dict, None, None]:
+    if max_date <= min_date:
+        raise ValueError('End date must be after start date.')
 
     headers = {}
     if api_key:
         # Specific to SMART catalogs
         headers['x-api-key'] = api_key
 
-    while stack:
-        url = stack.pop()
-        collection = requests.get(url, headers=headers).json()
-        # see if there's pages
-        with suppress(KeyError):
-            for link in collection['links']:
-                if link['rel'] == 'next':
-                    stack.append(link['href'])
-                    break
-        # iterate through items
-        for item in collection['features']:
-            yield item
+    date = min_date
+    catalog = Client.open(url, headers=headers)
+    delta = timedelta(days=1)
+    while date <= max_date:
+        print(date)  # DEBUG
+        results = catalog.search(collections=collections, datetime=[date, date + delta])
+        for item in results.get_items():
+            yield get_stac_item_self_link(item.to_dict()['links'])
+        date += delta
 
 
-def get_stac_item_self_link(links):
-    for link in links:
-        if link['rel'] == 'self':
-            return link['href']
-    raise ValueError('No self link found')
+def handle_posts(iter_func, collection, dry_run, *args, **kwargs):
+    client = get_client(dry_run)
+    with concurrent.futures.ThreadPoolExecutor(max_workers=32) as executor:
+        future_to_url = {
+            executor.submit(client.watch.post_stac_file, url, collection): url
+            for url in iter_func(*args, **kwargs)
+        }
```

Comment:
This is a problem. We need each thread to individually and concurrently use the iterator